src/functionaltests/java/com/ericsson/ei/subscriptions/repeatHandler/SubscriptionRepeatHandlerSteps.java

```diff
@@ -0,0 +1,166 @@
+import gherkin.deps.com.google.gson.JsonParser;
+import org.apache.commons.io.FileUtils;
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.junit.Ignore;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.MvcResult;
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+@Ignore
+@AutoConfigureMockMvc
+public class SubscriptionRepeatHandlerSteps extends FunctionalTestBase {
+    private static final String AGGREGATED_OBJECT_FILE_PATH = "src/functionaltests/resources/aggragatedObject.json";
+    private static final String EVENTS_FILE_PATH = "src/test/resources/TestExecutionTestEvents.json";
+    private static final String RULES_FILE_PATH = "src/test/resources/TestExecutionObjectRules.json";
+    private static final String REPEAT_FLAG_SUBSCRIPTION_COLLECTIONS_WITH_ONE_MATCH = "src/functionaltests/resources/subscriptionRepeatHandlerOneMatch.json";
+    private static final String REPEAT_FLAG_SUBSCRIPTION_COLLECTIONS_WITH_TWO_MATCH = "src/functionaltests/resources/subscriptionRepeatHandlerTwoMatch.json";
+    private static final String AGGREGATED_OBJECT_ID = "b46ef12d-25gb-4d7y-b9fd-8763re66de47";
+
+    private JSONObject subscriptionWithOneMatch;
+    private JSONObject subscriptionWithTwoMatch;
+    private String subscriptionStrWithOneMatch;
+    private String subscriptionStrWithTwoMatch;
+    private String aggregatedObject;
+    private String subscriptionIdMatchedAggrIdObjQuery;
+    private ObjectMapper mapper = new ObjectMapper();
+
+    @Value("${aggregated.collection.name}")
+    private String collectionName;
+
+    @Value("${spring.data.mongodb.database}")
+    private String dataBaseName;
+
+    @Value("${subscription.collection.repeatFlagHandlerName}")
+    private String repeatFlagHandlerCollection;
+
+    @Autowired
+    private MockMvc mockMvc;
+
+    @Autowired
+    private RmqHandler rmqHandler;
+
+    @Autowired
+    private ObjectHandler objectHandler;
+
+    @Autowired
+    private MongoDBHandler mongoDBHandler;
+
+    @Autowired
+    private RulesHandler rulesHandler;
+
+    @Autowired
+    private ISubscriptionService subscriptionService;
+
+    @Autowired
+    private RunSubscription runSubscription;
+
+    @Before
+    public void init() throws IOException, JSONException {
+        assertTrue(mongoDBHandler.insertDocument(dataBaseName, collectionName, getJSONFromFile(AGGREGATED_OBJECT_FILE_PATH).toString()));
```

Comment:
You can maybe move this out in a cucumber test step,, since init is not a test.

Reply:
e.g. "Given that Aggregated Object is created in database......................."

Reply:
fixed